docs: restructure and apply diataxis to OCI docs (#213)
canonical/ubuntu-cloud-docs: oci/oci-explanation/ubuntu-oci-container.rst

.. _ubuntu-oci-container-images:

Ubuntu OCI container images
===========================

The Open Container Initiative (OCI) establishes standards for constructing container 
images that can be reliably installed across a variety of compliant host environments.

Ubuntu’s `LTS Docker Image Portfolio <https://ubuntu.com/security/docker-images>`_ 
provides OCI-compliant images that receive stable security updates and predictable 
software updates, thus ensuring consistency in both maintenance schedule and operational 

> interfaces for the underlying software your software builds on.

Reviewer: The double use of "software" reads a bit strangely. I'd probably change the first one to something else.

Moreover, `Ubuntu-based containers <https://ubuntu.com/containers>`_ (like

> chiselled container images) all leverage the Ubuntu container image as their

Reviewer: As a newcomer, I don't know what a chiselled container image is. If we have a page that explains them we should link to it here.

Author: Y. The Chisel docs will be coming during the next cycle. For now we're documenting it within the Rockcraft docs. I'm adding a reference to it

> starting point, and thus also profit from its support and security commitments.

Reviewer: ```suggestion
starting point, and so also benefit from its support and security commitments.
```

Reviewer: Also, is there a page (from security, perhaps) that outlines the security commitments?

Actually now that I think about it, we might also want to find a different word than "support", since that has connotations of paid support.

Author: Y, adding a link to https://ubuntu.com/security/docker-images

Wrt the "support" word, I get what you're saying. In reality there's a notion of support for LTS too - Long Term Support - which is freemium and that's what we try to convey here.

The Ubuntu container image is built from a minimal rootfs tarball. This tarball

> is similar to `Ubuntu Base <https://wiki.ubuntu.com/Base>`_ 

Reviewer: These old wiki pages are going away "soon" :tm: and from what I've heard Ubuntu Base is mainly used internally. We will probably just want to remove the reference to Ubuntu Base altogether, otherwise we're expecting users to be familiar with it and chances are they won't be.

Author: Fair enough. Removing

> The Ubuntu container image is built from a minimal rootfs tarball. This tarball
> is similar to `Ubuntu Base <https://wiki.ubuntu.com/Base>`_ 
> but it already contains the modifications needed to make the rootfs suitable for 
> building OCI/Docker container images. The Ubuntu OCI rootfs tarballs are published

Reviewer: ```suggestion
The Ubuntu container image is built from a minimal rootfs tarball. This tarball
contains the modifications needed to make the rootfs suitable for 
building OCI/Docker container images. The Ubuntu OCI rootfs tarballs are published
```

.. _ubuntu-oci-container-images:

Ubuntu OCI container images
===========================

The `Open Container Initiative (OCI) <https://opencontainers.org/>`_ establishes standards for constructing container 
images that can be reliably installed across a variety of compliant host environments.

Ubuntu’s `LTS Docker Image Portfolio <https://ubuntu.com/security/docker-images>`_ 
provides OCI-compliant images that receive stable security updates and predictable 
software updates, ensuring consistency in both maintenance schedule and operational 
interfaces for the underlying foundation your software builds on.

> Moreover, `Ubuntu-based containers <https://ubuntu.com/containers>`_ (like

Reviewer 2: ```suggestion
Moreover, all `Ubuntu-based containers <https://ubuntu.com/containers>`_ (like
```

`chiselled container images <https://canonical-rockcraft.readthedocs-hosted.com/en/latest/explanation/chisel/>`_)

> all leverage the Ubuntu container image as their

Reviewer 2: ```suggestion
leverage the Ubuntu container image as their
```
